components/create-collective/sections/CreateCollectiveForm.js

```diff
@@ -0,0 +1,238 @@
+import { defaultBackgroundImage } from '../../../lib/constants/collectives';
+import CollectiveCategoryPicker from './CollectiveCategoryPicker';
+
+class CreateCollectiveForm extends React.Component {
+  static propTypes = {
+    host: PropTypes.object,
+    collective: PropTypes.object,
+    loading: PropTypes.bool,
+    onSubmit: PropTypes.func,
+    intl: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+
+    const collective = { ...(props.collective || {}) };
+    collective.slug = collective.slug ? collective.slug.replace(/.*\//, '') : '';
+
+    collective.backgroundImage = collective.backgroundImage || defaultBackgroundImage[collective.type];
+
+    this.state = {
+      collective,
+      checked: false,
+    };
+
+    this.categories = get(props.host, 'settings.apply.categories') || [];
+
+    if (this.categories.length === 1) {
+      this.state.collective.category = this.categories[0];
+    }
+
+    this.messages = defineMessages({
+      'tos.label': {
+        id: 'collective.tos.label',
+        defaultMessage: 'Terms of Service',
+      },
+    });
+  }
+
+  handleChange(fieldname, value) {
+    const collective = {};
+
+    collective[fieldname] = value;
+
+    this.setState({
+      collective: assign(this.state.collective, collective),
+    });
+  }
+
+  render() {
+    const { host, intl } = this.props;
+
+    const showForm = Boolean(this.categories.length === 0 || this.state.collective.category);
+
+    const initialValues = {
+      collectiveName: '',
+      collectiveDesc: '',
+      collectiveUrl: '',
+    };
+
+    const validate = values => {
+      const errors = {};
+
+      if (values.collectiveName.length > 10) {
+        errors.collectiveName = 'Please use fewer than 10 characters';
+      }
+
+      if (values.collectiveUrl.length < 5) {
+        errors.collectiveUrl = 'Please use more than 5 characters';
+      }
+
+      if (!values.collectiveDesc) {
+        errors.collectiveDesc = 'You need a description.';
+      }
+
+      return errors;
+    };
+
+    const submit = values => {
+      const { collectiveDesc, collectiveName, collectiveUrl } = values;
+      const collective = {
+        name: collectiveName,
+        description: collectiveDesc,
+        website: collectiveUrl,
+      };
+      assign(collective, this.state.collective);
+      this.props.onSubmit(collective);
+    };
+
+    return (
+      <div className="CreateCollectiveForm">
+        <div className="FormInputs">
+          {this.categories.length > 1 && (
+            <CollectiveCategoryPicker onChange={value => this.handleChange('category', value)} />
+          )}
+
+          {showForm && (
+            <div className="Form">
+              <Box p={4} m={4} style={{ border: '1px solid lightgrey', borderRadius: 8 }}>
+                <Formik
+                  validateOnChange={false}
+                  validateOnBlur={true}
```

Comment:
Since the validation is pretty much synchronous, I would validate `onChange`.
It's nice to know the name I'm picking is not valid while I'm still focusing the input.
Another way to fix this is by enforcing the limit size in the field.

Reply:
Interesting... I usually find it frustrating when I'm typing in a field and have not yet finished and it is already telling me what is wrong, but I can see from your perspective too how some people might find that helpful.

Reply:
Agree with @sbinlondon on this one. I've read from multiple sources ([here](https://ux.stackexchange.com/a/48269) for example) that `onChange` validation is not a good UX practice because it distracts users.

Some `onChange` validations do make sense tho, for example an asynchronous validation on the username to tell if it's unique or not.

That'd be a good topic to discuss in the design <> engineering call!

Reply:
Yep! On that note, @Betree, I've noticed that when I validate `onChange`, after I fill in the first field and click to the second, it immediately validates all the fields, even when they have nothing in them yet. Is there a way to only validate fields that have had their `initialValue` changed or would we have to add some extra logic ourselves?

Reply:
On that note, re: `onChange` validation, I also find it not great that Formik immediately validates all fields aftet you only fill in the first field and then move on to the second.

There seems to be some tension and discussion with Formik's users asking for the ability to only validate the field that has been touched, but it seems there's still no clear, easy answer? In this [issue](https://github.com/jaredpalmer/formik/issues/860) Jared mentions passing `validate` to each field individually... do you think that's an ok solution or what are you doing for your expense form?

Reply:
I didn't take this approach on the expense form because some validations depends on other fields (for example, validating the expense URL depends on the expense type).

If passing individual `validate` functions is a better fit for what you want to achieve feel free to use it, both approaches are good.